### ISSUE TITLE
DB connection refused

### ISSUE THREAD

**Issue by mesmith75** [ganga-devs/ganga#1792]
On AFS the database gives an error - something about a connection being refused:
```python
*** Welcome to Ganga ***
Version: 8.4.0 - DEV
Documentation and support: http://cern.ch/ganga
Type help() or help('index') for online help.

This is free software (GPL), and you are welcome to redistribute it
under certain conditions; type license() for details.


INFO     reading config file /afs/cern.ch/user/m/masmith/.gangarc
INFO     reading config file /afs/cern.ch/user/m/masmith/cmtuser/GANGA/GANGA_HEAD/install/ganga/ganga/GangaLHCb/LHCb.ini
INFO     reading config file /cvmfs/lhcb.cern.ch/lib/GangaConfig/config/8-0-0/GangaLHCb.ini
INFO     Using LHCbDirac version prod
Downloading...
From: https://cernbox.cern.ch/index.php/s/fv37NbLf24FpoWY/download
To: /afs/cern.ch/user/m/masmith/mongo.sif
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 174M/174M [00:01<00:00, 133MB/s]
INFO     Singularity gangaDB started on port: 58118
Traceback (most recent call last):
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/bin/ganga", line 90, in <module>
    GangaCore.Runtime.setupGanga()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/__init__.py", line 19, in setupGanga
    GangaCore.Runtime._prog.bootstrap(GangaCore.Runtime._prog.interactive)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/bootstrap.py", line 932, in bootstrap
    startUpRegistries()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/Repository_runtime.py", line 266, in startUpRegistries
    for n, k, d in bootstrap():
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/Repository_runtime.py", line 135, in bootstrap
    registry.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/GPIDev/Lib/Registry/PrepRegistry.py", line 29, in startup
    super(PrepRegistry, self).startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/DatabaseRegistry.py", line 558, in startup
    self.metadata.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/DatabaseRegistry.py", line 566, in startup
    self.repository.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 199, in startup
    self.update_index(True, True, True)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 522, in update_index
    objs = self._read_master_cache()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 484, in _read_master_cache
    for cache in master_cache:
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/cursor.py", line 1207, in next
    if len(self.__data) or self._refresh():
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/cursor.py", line 1100, in _refresh
    self.__session = self.__collection.database.client._ensure_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1816, in _ensure_session
    return self.__start_session(True, causal_consistency=False)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1766, in __start_session
    server_session = self._get_server_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1802, in _get_server_session
    return self._topology.get_server_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/topology.py", line 488, in get_server_session
    None)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/topology.py", line 217, in _select_servers_loop
    (self._error_message(selector), timeout, self.description))
pymongo.errors.ServerSelectionTimeoutError: localhost:58118: [Errno 111] Connection refused, Timeout: 10.0s, Topology Description: <TopologyDescription id: 5f81b8a55d9d8eefb2e293df, topology_type: Single, servers: [<ServerDescription ('localhost', 58118) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:58118: [Errno 111] Connection refused',)>]>
INFO     Stopping Job processing before shutting down Repositories
INFO     Shutting Down Ganga Repositories
INFO     Registry Shutdown
```
This occurs if I start ganga on lxplus, or locally with the gangadir on AFS. If I start it locally with the gangadir local it works fine. I guess there is some permission issue?

**Comment by egede** [comment]
I reproduce this problem. As a help I attach the `gangadir/logs/mongod-ganga.log` file.

```
{"t":{"$date":"2020-10-12T05:03:23.586+02:00"},"s":"I",  "c":"CONTROL",  "id":23285,   "ctx":"main","msg":"Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'"}
{"t":{"$date":"2020-10-12T05:03:23.590+02:00"},"s":"W",  "c":"ASIO",     "id":22601,   "ctx":"main","msg":"No TransportLayer configured during NetworkInterface startup"}
{"t":{"$date":"2020-10-12T05:03:23.591+02:00"},"s":"I",  "c":"NETWORK",  "id":4648601, "ctx":"main","msg":"Implicit TCP FastOpen unavailable. If TCP FastOpen is required, set tcpFastOpenServer, tcpFastOpenClient, and tcpFastOpenQueueSize."}
{"t":{"$date":"2020-10-12T05:03:23.591+02:00"},"s":"I",  "c":"STORAGE",  "id":4615611, "ctx":"initandlisten","msg":"MongoDB starting","attr":{"pid":2713,"port":40239,"dbPath":"/data/db","architecture":"64-bit","host":"lxplus7117.cern.ch"}}
{"t":{"$date":"2020-10-12T05:03:23.591+02:00"},"s":"I",  "c":"CONTROL",  "id":23403,   "ctx":"initandlisten","msg":"Build Info","attr":{"buildInfo":{"version":"4.4.0","gitVersion":"563487e100c4215e2dce98d0af2a6a5a2d67c5cf","openSSLVersion":"OpenSSL 1.1.1  11 Sep 2018","modules":[],"allocator":"tcmalloc","environment":{"distmod":"ubuntu1804","distarch":"x86_64","target_arch":"x86_64"}}}}
{"t":{"$date":"2020-10-12T05:03:23.591+02:00"},"s":"I",  "c":"CONTROL",  "id":51765,   "ctx":"initandlisten","msg":"Operating System","attr":{"os":{"name":"Ubuntu","version":"18.04"}}}
{"t":{"$date":"2020-10-12T05:03:23.591+02:00"},"s":"I",  "c":"CONTROL",  "id":21951,   "ctx":"initandlisten","msg":"Options set by command line","attr":{"options":{"net":{"bindIp":"*","port":40239},"systemLog":{"destination":"file","path":"/afs/cern.ch/user/u/uegede/gangadir/logs/mongod-ganga.log"}}}}
{"t":{"$date":"2020-10-12T05:03:23.601+02:00"},"s":"E",  "c":"STORAGE",  "id":20557,   "ctx":"initandlisten","msg":"DBException in initAndListen, terminating","attr":{"error":"NonExistentPath: Data directory /data/db not found. Create the missing directory or specify another path using (1) the --dbpath command line option, or (2) by adding the 'storage.dbPath' option in the configuration file."}}
{"t":{"$date":"2020-10-12T05:03:23.601+02:00"},"s":"I",  "c":"NETWORK",  "id":20562,   "ctx":"initandlisten","msg":"Shutdown: going to close listening sockets"}
{"t":{"$date":"2020-10-12T05:03:23.601+02:00"},"s":"I",  "c":"-",        "id":20520,   "ctx":"initandlisten","msg":"Stopping further Flow Control ticket acquisitions."}
{"t":{"$date":"2020-10-12T05:03:23.601+02:00"},"s":"I",  "c":"CONTROL",  "id":20565,   "ctx":"initandlisten","msg":"Now exiting"}
{"t":{"$date":"2020-10-12T05:03:23.601+02:00"},"s":"I",  "c":"CONTROL",  "id":23138,   "ctx":"initandlisten","msg":"Shutting down","attr":{"exitCode":100}}
```
Based on the `Data directory /data/db not found` message, I tried to create the directory `gangadir/data/db`. Then starting `ganga` again gave the message

```
ERROR    Singularity container could not start because of: NonExistentPath: Data directory /data/db not found. Create the missing directory or specify another path using (1) the --dbpath command line option, or (2) by adding the 'storage.dbPath' option in the configuration file.
INFO     Singularity gangaDB started on port: 40239
```
which however left a running `mongodb`. Starting `ganga` a third time gave a bunch of nasty messages at startup

```
ERROR    Singularity container could not start because of: 95
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
```
but a working session. However after a start/stop of `ganga` a few times, I see the following

```
[uegede@lxplus7117]~% ps -elf
F S UID        PID  PPID  C PRI  NI ADDR SZ WCHAN  STIME TTY          TIME CMD
4 S uegede    7143     1  0  80   0 - 101371 futex_ 05:10 pts/65  00:00:00 Singularity runtime parent
4 S uegede    7376  7143  3  80   0 - 404156 futex_ 05:10 pts/65  00:00:14 mongod --port 40239 --logpath /afs/cern.ch/user/u/uegede/gangadir/logs/mongod-ganga.log --bind_ip_all
0 S uegede   15171     1  1  80   0 - 108308 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15239     1  0  80   0 - 87762 futex_ 05:17 pts/65   00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15313     1  0  80   0 - 141076 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15352     1  0  80   0 - 122707 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15390     1  0  80   0 - 143253 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15433     1  0  80   0 - 124692 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15480     1  1  80   0 - 87826 futex_ 05:17 pts/65   00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15522     1  0  80   0 - 143125 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15571     1  1  80   0 - 106323 futex_ 05:17 pts/65  00:00:00 singularity run --bind /afs/cern.ch/user/u/uegede/gangadir/data:/data /afs/cern.ch/user/u/uegede/gangadir/mongo.sif mongod --port 40239 
0 S uegede   15572 15171  7  80   0 - 289508 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-305586479/root /tmp/uegede/rootfs-305586479/archive-745030338
0 S uegede   15708 15390  9  80   0 - 289535 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-351942488/root /tmp/uegede/rootfs-351942488/archive-098471127
0 S uegede   15709 15313  9  80   0 - 289535 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-860467366/root /tmp/uegede/rootfs-860467366/archive-150829005
0 S uegede   15710 15480 10  80   0 - 289535 futex_ 05:17 pts/65  00:00:04 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-177546888/root /tmp/uegede/rootfs-177546888/archive-051516487
0 S uegede   15711 15352 10  80   0 - 289535 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-049079055/root /tmp/uegede/rootfs-049079055/archive-798026018
0 S uegede   15712 15433  9  80   0 - 289535 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-773922398/root /tmp/uegede/rootfs-773922398/archive-717042725
0 S uegede   15788 15239 10  80   0 - 289536 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-407870242/root /tmp/uegede/rootfs-407870242/archive-453023257
0 S uegede   15814 15522 11  80   0 - 289534 futex_ 05:17 pts/65  00:00:04 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-492759611/root /tmp/uegede/rootfs-492759611/archive-635387486
0 S uegede   16177 15571 13  80   0 - 289500 futex_ 05:17 pts/65  00:00:03 /sbin/unsquashfs -user-xattrs -f -d /tmp/uegede/rootfs-684815201/root /tmp/uegede/rootfs-684815201/archive-126055756
4 R uegede   16272 25074  0  80   0 - 41443 -      05:18 pts/65   00:00:00 ps -elf
5 S uegede   25069 24941  0  80   0 - 45954 poll_s 04:47 ?        00:00:00 sshd: uegede@pts/65
0 S uegede   25074 25069  0  80   0 - 35612 sigsus 04:47 pts/65   00:00:00 -zsh
```
indicating that there indeed is a problem detecting running `singularity` containers (or a failure to shut them down).

**Comment by egede** [comment]
I think one problem might be related to the 1 second delay at https://github.com/ganga-devs/ganga/blob/d4915116fd65aff753a29d332d9578e3f29e4185/ganga/GangaCore/Core/GangaRepository/container_controllers.py#L269 If this delay is not long enough it seems like the code tries to start another container. The `ganga` process then creates multiple orphan singularity processes. Below is the output from a fresh login session to a new machine. As you can see the start-up of Ganga fails completely the first time but then after some warnings work the second time.

```
[uegede@lxplus784]~% ps
  PID TTY          TIME CMD
13183 pts/69   00:00:00 zsh
13271 pts/69   00:00:00 ps
[uegede@lxplus784]~% ganga --ganga-version DEV

*** Welcome to Ganga ***
Version: 8.4.0 - DEV
Documentation and support: http://cern.ch/ganga
Type help() or help('index') for online help.

This is free software (GPL), and you are welcome to redistribute it
under certain conditions; type license() for details.


INFO     reading config file /afs/cern.ch/user/u/uegede/.gangarc
INFO     reading config file /cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaLHCb/LHCb.ini
INFO     reading config file /cvmfs/lhcb.cern.ch/lib/GangaConfig/config/8-0-0/GangaLHCb.ini
INFO     Using LHCbDirac version prod
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
Traceback (most recent call last):
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/bin/ganga", line 90, in <module>
    GangaCore.Runtime.setupGanga()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/__init__.py", line 19, in setupGanga
    GangaCore.Runtime._prog.bootstrap(GangaCore.Runtime._prog.interactive)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/bootstrap.py", line 932, in bootstrap
    startUpRegistries()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/Repository_runtime.py", line 266, in startUpRegistries
    for n, k, d in bootstrap():
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Runtime/Repository_runtime.py", line 135, in bootstrap
    registry.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/GPIDev/Lib/Registry/PrepRegistry.py", line 29, in startup
    super(PrepRegistry, self).startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/DatabaseRegistry.py", line 558, in startup
    self.metadata.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/DatabaseRegistry.py", line 566, in startup
    self.repository.startup()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 199, in startup
    self.update_index(True, True, True)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 522, in update_index
    objs = self._read_master_cache()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaCore/Core/GangaRepository/GangaRepositoryDatabase.py", line 484, in _read_master_cache
    for cache in master_cache:
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/cursor.py", line 1207, in next
    if len(self.__data) or self._refresh():
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/cursor.py", line 1100, in _refresh
    self.__session = self.__collection.database.client._ensure_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1816, in _ensure_session
    return self.__start_session(True, causal_consistency=False)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1766, in __start_session
    server_session = self._get_server_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1802, in _get_server_session
    return self._topology.get_server_session()
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/topology.py", line 488, in get_server_session
    None)
  File "/cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/pymongo/topology.py", line 217, in _select_servers_loop
    (self._error_message(selector), timeout, self.description))
pymongo.errors.ServerSelectionTimeoutError: localhost:40239: [Errno 111] Connection refused, Timeout: 10.0s, Topology Description: <TopologyDescription id: 5f83d2b651682ac0963964b9, topology_type: Single, servers: [<ServerDescription ('localhost', 40239) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:40239: [Errno 111] Connection refused',)>]>
INFO     Stopping Job processing before shutting down Repositories
INFO     Shutting Down Ganga Repositories
INFO     Registry Shutdown
[uegede@lxplus784]~% 
[uegede@lxplus784]~% 
[uegede@lxplus784]~% ganga --ganga-version DEV

*** Welcome to Ganga ***
Version: 8.4.0 - DEV
Documentation and support: http://cern.ch/ganga
Type help() or help('index') for online help.

This is free software (GPL), and you are welcome to redistribute it
under certain conditions; type license() for details.


INFO     reading config file /afs/cern.ch/user/u/uegede/.gangarc
INFO     reading config file /cvmfs/ganga.cern.ch/Ganga/install/DEV/lib/python3.6/site-packages/ganga/GangaLHCb/LHCb.ini
INFO     reading config file /cvmfs/lhcb.cern.ch/lib/GangaConfig/config/8-0-0/GangaLHCb.ini
INFO     Using LHCbDirac version prod
ERROR    Singularity container could not start because of: 95
ERROR    Singularity container could not start because of: 95
ERROR    Singularity container could not start because of: 95
ERROR    Singularity container could not start because of: 95
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
cat: /proc/14086/environ: No such file or directory
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
cat: /proc/14263/environ: No such file or directory
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
cat: /proc/14354/environ: No such file or directory
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
ERROR    Singularity container could not start because of: {'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}
INFO     Singularity gangaDB started on port: 40239
 === Welcome to Ganga on CVMFS. In case of problems contact lhcb-distributed-analysis@cern.ch === 

[05:51:58]                                                                                                                                                                                                         
Ganga In [1]: jobs                                                                                                                                                                                                 
Ganga Out [1]: 
Registry Slice: jobs (2 objects)
--------------
    fqid |    status |      name | subjobs |    application |        backend |                             backend.actualCE |                       comment |  subjob status 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
       0 | completed |           |         |     Executable |      Localhost |                           lxplus7117.cern.ch |                               |        0/0/0/0 
       1 |       new |           |         |     Executable |      Localhost |                                              |                               |        0/0/0/0 


[05:51:58]                                                                                                                                                                                                         
Ganga In [2]:              
```
The 95 error reported is in the logfile coresponding to 
```
{"t":{"$date":"2020-10-12T05:51:31.396+02:00"},"s":"E",  "c":"STORAGE",  "id":22435,   "ctx":"initandlisten","msg":"WiredTiger error","attr":{"error":95,"message":"[1602474691:396829][13687:0x7f1e72ff1a80], connection: __posix_std_fallocate, 58: /data/db/journal/WiredTigerTmplog.0000000001: fallocate:: Operation not supported"}}
```

**Comment by egede** [comment]
When `ganga` tries to identify if `mongodb` is running inside the container, it looks for the environment variables of processes and will try to identify
```
SINGULARITY_CONTAINER=.../gangadir/singularity.sif
```
but at CERN I see the following

```
SINGULARITY_CONTAINER=/tmp/uegede/rootfs-083973340/root
```
so Singularity starts in a sandbox environment by default and the check fails.

**Comment by egede** [comment]
It seems like most of the problems here are solved by #1795 .

**Comment by egede** [comment]
https://cern.service-now.com/service-portal?id=ticket&table=incident&n=INC2574151

Running singularity containers is broken on many lxplus nodes. The
example below illustrates the problem.
 
*Broken node:*
 ```
[uegede@lxplus717]~% /usr/bin/singularity run docker://alpine:latest
echo 'SUCCESS'
INFO:    Using cached SIF image
INFO:    Convert SIF file to sandbox...
INFO:    Cleaning up image...
FATAL:   container creation failed: mount
/afs/cern.ch/user/u/uegede->/var/singularity/mnt/session/afs/cern.ch/user/u/uegede
error: while mounting /afs/cern.ch/user/u/uegede: destination
/var/singularity/mnt/session/afs/cern.ch/user/u/uegede doesn't exist in
container
```

*Functional node:*
``` 
[uegede@lxplus752]~% /usr/bin/singularity run docker://alpine:latest
echo 'SUCCESS'
INFO:    Using cached SIF image
INFO:    Convert SIF file to sandbox...
SUCCESS
INFO:    Cleaning up image...
``` 
Regards,
 
        Ulrik Egede
